scio-extra/src/main/scala/com/spotify/scio/extra/Sparkey.scala

```diff
@@ -0,0 +1,166 @@
+
+import java.io._
+import java.net.URI
+import java.nio.ByteBuffer
+import java.nio.file.{Paths, Files => JFiles}
+
+import com.google.common.io.Files
+import com.spotify.scio.ScioContext
+import com.spotify.scio.values.{SCollection, SideInput}
+import com.spotify.sparkey.{CompressionType, SparkeyReader, SparkeyWriter, Sparkey => JSparkey}
+import org.apache.beam.runners.dataflow.options.DataflowPipelineOptions
+import org.apache.beam.sdk.options.PipelineOptionsFactory
+import org.apache.beam.sdk.transforms.{DoFn, View}
+import org.apache.beam.sdk.util.GcsUtil.GcsUtilFactory
+import org.apache.beam.sdk.util.gcsfs.GcsPath
+import org.apache.beam.sdk.values.PCollectionView
+
+import scala.collection.JavaConverters._
+import scala.util.Try
+
+object Sparkey {
+
+  implicit class SparkeyScioContext(val self: ScioContext) {
+    def sparkeySideInput(url: SparkeyUri): SideInput[SparkeyReader] = {
+      val view = self.parallelize(Seq(url)).applyInternal(View.asSingleton())
+      new SparkeySideInput(view)
+    }
+  }
+
+  implicit class SCollectionWithSparkeyWriter[K, V](val self: SCollection[(K, V)])
+                                                   (implicit ev1: K <:< String, ev2: V <:< String) {
+    def asSparkey(uri: SparkeyUri): SCollection[SparkeyUri] = self.transform { in =>
+      in.groupBy(_ => ())
+        .map { case (_, iter) =>
+          val writer = uri.getWriter()
+          val it = iter.iterator
+          while (it.hasNext) {
+            val kv = it.next()
+            writer.put(kv._1.toString, kv._2.toString)
+          }
+          writer.close()
+          uri
+        }
+    }
+
+    def asSparkey: SCollection[SparkeyUri] = {
+      val tempLocation = Try(self.context.optionsAs[DataflowPipelineOptions].getGcpTempLocation)
+        .getOrElse(self.context.options.getTempLocation)
+      this.asSparkey(SparkeyUri(tempLocation))
+    }
+  }
+
+  implicit class SparkeySCollection(val self: SCollection[SparkeyUri]) {
+    def asSparkeySideInput(): SideInput[SparkeyReader] = {
+      val view = self.applyInternal(View.asSingleton())
+      new SparkeySideInput(view)
+    }
+  }
+
+  implicit class SCollectionToSparkeySideInput(val self: SCollection[(String, String)]) {
+    def asSparkeySideInput(): SideInput[SparkeyReader] = self.asSparkey.asSparkeySideInput()
+  }
+
+  implicit class Reader(val self: SparkeyReader) {
+    def get(key: String): Option[String] = Option(self.getAsString(key))
+
+    def toStream(): Stream[(String, String)] = self.iterator().asScala.toStream.map { e =>
+      (e.getKeyAsString, e.getValueAsString)
+    }
+  }
+
+  private[scio] class SparkeySideInput(val view: PCollectionView[SparkeyUri])
+    extends SideInput[SparkeyReader] {
+    override def get[I, O](context: DoFn[I, O]#ProcessContext): SparkeyReader =
+      SparkeyUri(context.sideInput(view).path).getReader()
+  }
+}
+
+trait SparkeyUri extends Serializable {
+  val path: String
+  def getWriter(): WrappedSparkeyWriter
```

Comment:
Make `getWriter` private? The user shouldn't use it directly.
Also override `.toString` so `SparkeyUri` can be printed or saved as text file.